content/blog/oauth-oidc.md

```diff
@@ -0,0 +1,88 @@
+* the authorization server delivers an access token to the client
+* the client uses the access token to request the resource server
+
+The different authorization flows target different use cases, like mobile applications, single page web applications, smart TVs...
+
+## OIDC
+
+[OpenID Connect](https://openid.net/connect/) builds upon OAuth 2.0 to provide an authentication layer. Why use OIDC instead of OAuth directly though? After all, getting an access token from the authorition server should be enough of a proof that the user is authenticated?
+
+Let's consider this scenario:
+* the user is connected to service A, a resource server and authorization server
+* services B and C use OAuth access to service A for authentication: they redirect the user to service A to get an access token, then test the access token can actually access the resources
+* the user connects to B and C using service A, so B and C recieved access token for the user's ressources on A
+* now, the malicious admin of service C wants to access service B. They start the OAuth authoization to service B, but at the point where they are redirected to A, and must return the new access token, they instead provide the token they already have for the user
+* B checks that the token is valid, and lets them in
+
+<details>
+For people who like to understand a bit more, this is one of the reasons the OAuth implicit flow is deprecated, that scenario wouldn't work under the authorization code flow
+</details>
+This is the fundamental misunderstanding here: a well behaving OAuth client will keep the access token safe, because it needs it to access resources. But there is no guarantee that it was delivered specifically for this client, nor that the user actually gave consent during this process. A misbehaving application could just use it as an authentication proof anywhere else. And we're not even getting into the issue of applications using your Github or Twitter accounts supposedly only for authentication, but requiring write access to everything.
+
+That is why the OIDC protocol was designed: providing a safer delegate authentication mechanism. Essentially, when a service requests that a user authenticates with an OIDC service, it will perform an OAuth authorization flow where the response contains an ID token. The ID token is a [JSON Web Token](https://datatracker.ietf.org/doc/html/rfc7519), which contains [a specific set of claims](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) used to identify the user. Some of those claims are required and carry the security of the system, like `aud` indicating for which serice this token was created, `auth_time` for the exact time when the authentication was performed, or the `nonce` claim containing a value provided in the authentication request (if it's not the same as what was requested, the token is considered invalid).
```

Comment:
```suggestion
That is why the OIDC protocol was designed: providing a safer delegate authentication mechanism. Essentially, when a service requests that a user authenticates with an OIDC service, it will perform an OAuth authorization flow where the response contains an ID token. The ID token is a [JSON Web Token](https://datatracker.ietf.org/doc/html/rfc7519), which contains [a specific set of claims](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) used to identify the user. Some of those claims are required and carry the security of the system, like `aud` indicating for which service this token was created, `auth_time` for the exact time when the authentication was performed, or the `nonce` claim containing a value provided in the authentication request (if it's not the same as what was requested, the token is considered invalid).
```

Reply:
What does "carry the security" mean here?